enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -615,51 +622,45 @@ func (s *COWStore) initDirtyChunk(offset int64, size int64) (ogChunk *Mmap, newC
 	return ogChunk, newChunk, nil
 }
 
-type eagerFetchData struct {
-	// Chunk offset to eagerly fetch
-	offset int64
-}
-
 func (s *COWStore) eagerFetchNextChunks(offset int64) {
 	currentOffset := offset + s.chunkSizeBytes
 	for i := 0; i < numChunksToEagerFetch; i++ {
-		s.sendNonBlockingEagerFetch(currentOffset)
+		s.eagerFetchStack.Push(currentOffset)
 		currentOffset += s.chunkSizeBytes
 	}
 }
 
-func (s *COWStore) sendNonBlockingEagerFetch(offset int64) {
-	select {
-	case s.eagerFetchChan <- &eagerFetchData{offset: offset}:
-	default:
-	}
-}
-
 func (s *COWStore) eagerFetchChunksInBackground() {
 	rateLimiter := rate.NewLimiter(rate.Limit(*maxEagerFetchesPerSec), 1)
 	eg := &errgroup.Group{}
 	eg.SetLimit(*eagerFetchConcurrency)
 	defer eg.Wait()
 
+	ctx, cancel := context.WithCancel(s.ctx)
+	defer cancel()
+	go func() {
+		<-s.quitChan
+		cancel()
+	}()
```

Comment:
is s.ctx cancelled when quitChan is called? Seems like if it were you would no longer need this. (and maybe not even need quitChan either since you could instead wait for <-s.ctx.Done() instead)

Reply:
We don't cancel `s.ctx` in the place where `quitChan` is called. I'd prefer not to change that right now because there may be some unintended consequences there that I'm not seeing.